File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/lua/KissOfDuplication.lua
function GetName()
  return "Kiss of Duplication"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { Attacking }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == Attacking and Roll(0, 99) < 5 then
    SummonMinionCopy(param1, param1:GetName())
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/KissOfDuplication.cs
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class KissOfDuplication : Affliction
    {
        public override string Name { get { return "Kiss of Duplication"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.Attacking, 
                };
            }
        }

        protected override void OnAttacking(ICharacter attacker, ICharacter target, ISpell spell, ref double damage, ref double modifier)
        {
            if (RollPercentage(5))
            {
                SummonMinionCopy(target, target.Name);
            }
        }
    }
}